cache/token_test.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2025 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache_test
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/fluxcd/pkg/cache"
+)
+
+type testToken struct {
+	duration time.Duration
+}
```

Comment:
Can you please make this test case closer to the actual implementation in let's say source-controller & GitHub App.

I expect for this type to hold a GitHub token and also the kind, name, namespace. Then a function that composes the cache key.

An implementation here would help reviewing and also help consumers of this package.

Reply:
In order not to duplicate all this stuff here and in the actual implementation I created the following combo PR based on this one: https://github.com/fluxcd/pkg/pull/871